### ISSUE TITLE
Run method never gets called

### ISSUE THREAD

**Issue by sverin** [logstash/logstash-logback-encoder#71]
I'm downloaded the source code, build the jar and deployed it on a IBM domino server.
Then I configured it to use the LogstashTcpSocketAppender.

But no logs are sent to logstash.

It seems to be an issue in the run loop. I've added some comments  and it never enters the while loop. It seems to be interrupted before.

**Comment by philsttr** [comment]
Hi @sverin 

Can you post your logback configuration?

Generally, when the run method is not called, it indicates a problem with the configuration.  

Also, logback will print configuration errors to the console on startup, so you might be able to find hints to what the problem is there.

If you want to dive further into the code, take a look at ['AbstractLogstashTcpSocketAppender.start'](https://github.com/logstash/logstash-logback-encoder/blob/master/src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java#L192).  You'll notice that the errorCount has to be == 0 for the task to be submitted to the executor service.  The task must be submitted to the executorService for the run method to be called.

**Comment by sverin** [comment]
I looked at the AbstractLogstashTcpSocketAppender file and added some console logs in the run method. The run method is called but something happens in the while-loop. But no more information on the console.
```
public final void run() {
        try {
        	System.out.println("run start");
            while (!Thread.currentThread().isInterrupted()) {
            	System.out.println("Trying to connect...");
```
 The "run start" message is printed, but not "Trying to connect...", som it seems that the thread is interuppted somehow...

**Comment by sverin** [comment]
I forget the configuration file:

```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
  	
  	<appender name="SASU" class="net.logstash.logback.appender.LogstashTcpSocketAppender">
		<remoteHost>...</remoteHost>
		<port>...</port>
		<encoder class="net.logstash.logback.encoder.LogstashEncoder" />
  	</appender>
 
	<root level="DEBUG">
		<appender-ref ref="SASU" />
	</root>

</configuration>
```

**Comment by philsttr** [comment]
Are you just running a quick process that logs a few messages and exits?

If so, then the process might not be surviving long enough for the thread to pick up the event from the queue to send it over the wire.

**Comment by zeldigas** [comment]
I've also faced that problem. During debug I found that the reason is that in my project I have declared dependency from logback-classic with version 1.0.13.
There is an interface change between 1.0.13 and 1.1.x in constructor of DefaultSocketConnector (int arguments replaced with long) and due to that NoSuchMethod exception occured.
I understand that you defined version in pom.xml, but it would be great if run catch not only InterruptedExceptions but another throwables and add error messages to logback internal status collector
